code-challenges.test.js

```diff
@@ -10,43 +10,164 @@
+  const secretCodeWord1 = "Lackadaisical";
+  // Expected output: "L4ck4d41s1c4l"
+  const secretCodeWord2 = "Gobbledygook";
+  // Expected output: "G0bbl3dyg00k"
+  const secretCodeWord3 = "Eccentric";
+  // Expected output: "3cc3ntr1c"
+  it("takes in a string and returns a string with a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
+    expect(transcoder(secretCodeWord1)).toEqual("L4ck4d41s1c4l");
+    expect(transcoder(secretCodeWord2)).toEqual("G0bbl3dyg00k");
+    expect(transcoder(secretCodeWord3)).toEqual("3cc3ntr1c");
+  });
+});
+
+// ReferenceError: transcoder is not defined
 
 // b) Create the function that makes the test pass.
 
+// pseudo: (Gene, if you're reading this, I'm sorry for this pseudo lol.)
+
+// create a function named transcoder that take in a string as an argument
+// within our function we are going to call on the .replaceAll() method for each letter we are wanting to replace using a regular expression (regex) for each letter. .replaceAll() is a built in method for JS that takes in a value that we want to replace, and a replacement for that value. This replacement can be a function or a string.
+// since .replaceAll() can only take in one replacement, in order to replace all for values in our prompt, we would have to make an object within our function for our replacement to reference in order to replace all instances of the given values. This is a totally valid way to think about this problem.
+// im not going to do that
+// instead of creating an object, and passing a function into our .replaceAll(), i am just going to call on .replaceAll() for each value that i want to replace with one replacement per value.
+// I know what you're thinking "CJ, are you really going to call on the same method 4 times in your function?" Yes I am, here's why; i think it will read better, and i think it will be just as performant as the creating an object technique. Specifically given in our prompt we are only replacing 4 values, this pushes me to do the solution the way that i am, given a more complex prompt could require a refactor here.
+// So, each .replaceAll() will take a regex for each letter, and the replacement for that letter in a string. string.replaceAll(/a/gi, "4").replaceAll(/e/gi, "3").replaceAll(/i/gi, "1").replaceAll(/o/gi, "0")
+// regex is object in JS that we can use to match text with a pattern the data within the "//" is the value we are wanting to replace. The 'g' after our /a/ is required when using regex with .replaceAll(, it indicates global, which means that we are looking at every possible instance of that pattern, not just the first. The 'i' indicates ignoreCase which will account for both upper and lower case instances of the pattern.
+// return original string with these characters swapped: "a" to "4", "e" to "3", "i" to "1", and "o" to "0"
+// this could be one line, but for readability we are going to have each .replaceAll() have its own line.
+
```

Comment:
Well now that I made it through that pseudocode, I like this approach. For readability, I think it makes sense to have it as is. You could use replace since you have the g flag but I like the replaceAll because it leaves no doubt that this is working on every instance of that letter. Nicely done! 💯